### PR TITLE
Notify 473

### PR DESCRIPTION
**Pull request by A-Shumway42**
Added:

- Configurations to `jest.config.js` for javascript code coverage.
- Coverage directory to git ignore
- Using <https://github.com/marketplace/actions/jest-coverage-report> for code coverage output on PR.
- Using <https://github.com/jwalton/gh-find-current-pr> action to find current PR. By default, the `push` workflow doesn't know which PR to attach to.

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-permissions:
-  contents: read
+on:
+  push:
+    permissions:
+      contents: read
+  pull_request_target:
+    types:
+      - opened
+    permissions:
```
Comment (stvnrlly) [inline .github/workflows/checks.yml:10]: Could we move these permissions down to the `build` job, similar to what's in [this example](https://github.com/marketplace/actions/jest-coverage-report#pull-request-number)?

**Review by stvnrlly — CHANGES_REQUESTED** [review]
Just a matter of narrowing the permissions, otherwise 👍🏻 .

Looking at the comment that the action makes, I'm actually wondering if we want that. It's not clear from the docs that this is supported, but I think I'm more interested in the failing test than the comment-report. Part of it is that the JS is a smaller part of the codebase, so that being the only report could be confusing.

If there's a way to get the CI result without the comment, let's try that for now. Not granting the permissions might be a way to do that, if it's not otherwise supported.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (A-Shumway42) [reply .github/workflows/checks.yml:10]: I'll try that out and see if it works.

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -2,9 +2,6 @@ name: Run checks
 
 on: [push]
 
-permissions:
```
Comment (stvnrlly) [inline .github/workflows/checks.yml:5]: This permission should stay as it's needed for other jobs. I don't _think_ that'll fix the test failure, but 🤷🏻

**Review by stvnrlly — CHANGES_REQUESTED** [review]

**Review by stvnrlly — APPROVED** [review]

**Pull request merged**